README.md

```diff
@@ -14,54 +16,59 @@ Run `rails g async_request`
 
 ``` ruby
-class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
-
-  def my_compute_heavy_endpoint
-    id = execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
-    render json: { id: id, url: async_request.job_url(id) }, status: 202
+class SomeController < ApplicationController
+  def my_compute_heavy_endpoint_option_1
+    _job, token = AsyncRequest::Job.execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
+    render json: { token: token, url: async_request.job_url }, status: :accepted
+  end
+
+  def my_compute_heavy_endpoint_option_2
+    _job, token = AsyncRequest::Job.execute_async(MyComputeHeavyWorker, { some: 'args' }, 'another arg')
+    render json: { token: token }, status: :accepted, location: async_request.job_url
+  end
+end
+
+class MyComputeHeavyWorker
+  def execute(options, string)
+    # Perform heavy task
+    [:ok, { message: 'success' }.to_json] # returning 200 is valid too
   end
 end
 ```
-This will enqueue a Sidekiq task that will call `MyComputeHeavyWorker.new.execute` with the args passed as parameters to `execute_async`.
 
-The client can then make a GET request to the returned URL. If the job's completed, it will get a response code and a json body. Otherwise, a `202` code will be returned
+This will enqueue a [Sidekiq worker](https://github.com/Wolox/async-requests/blob/master/app/workers/async_request/job_processor.rb) that will call `MyComputeHeavyWorker.new.execute` with the args passed as parameters to `execute_async`.
+
+The client can then make a GET request to the returned URL sending the `token` in a `X-JOB-AUTHORIZATION` header (configurable). If the job's completed, it will get a response code and a json body. Otherwise, a `202 (accepted)` code will be returned.
+
+`MyComputeHeavyWorker` must return an array with two components. The first one has be the status code (number or Rails symbol), the second one has be the response that will be sent to the client. Take into account that the response must be a string because it will be stored in the database.
 
-`MyComputeHeavyWorker` must return an array with two components. The first one must be the status code, the second one must be the response that will be sent to the client. Take into account that the response must respond to `to_json`.
+### Configuration
 
+There are some aspects of the gem you can configure:
+
+``` ruby
+# config/initializers/async_request.rb
+AsyncRequest.configure do |config|
+  config.sign_algorithm = 'HS256' # This is the default, valid algorithms: HS256 and RS256
+  config.request_header_key = 'X-JOB-AUTHORIZATION' # This is the default
+  config.encode_key = Rails.application.secrets.secret_key_base
+  config.decode_key = Rails.application.secrets.secret_key_base
+  config.token_expiration = Time.zone.now + 1.day
```

Comment:
shouldn't this be a proc ? if not ALL token will expire 1 day after the server is initialized, right ?